src/sentry/search/events/datasets/functions.py

```diff
@@ -0,0 +1,359 @@
+        "min",
+        "max",
+        "avg",
+        "sum",
+        "worst",
+        "examples",
+    }
+
+    def __init__(self, builder: builder.QueryBuilder):
+        self.builder = builder
+
+    @property
+    def search_filter_converter(
+        self,
+    ) -> Mapping[str, Callable[[SearchFilter], Optional[WhereType]]]:
+        return {
+            "message": self._message_filter_converter,
+            PROJECT_ALIAS: self._project_slug_filter_converter,
+            PROJECT_NAME_ALIAS: self._project_slug_filter_converter,
+        }
+
+    def _message_filter_converter(self, search_filter: SearchFilter) -> Optional[WhereType]:
+        value = search_filter.value.value
+        if search_filter.value.is_wildcard():
+            # XXX: We don't want the '^$' values at the beginning and end of
+            # the regex since we want to find the pattern anywhere in the
+            # message. Strip off here
+            value = search_filter.value.value[1:-1]
+            return Condition(
+                Function("match", [self.builder.column("message"), f"(?i){value}"]),
+                Op(search_filter.operator),
+                1,
+            )
+        elif value == "":
+            operator = Op.EQ if search_filter.operator == "=" else Op.NEQ
+            return Condition(
+                Function("equals", [self.builder.column("message"), value]), operator, 1
+            )
+        else:
+            if search_filter.is_in_filter:
+                return Condition(
+                    self.builder.column("message"),
+                    Op(search_filter.operator),
+                    value,
+                )
+
+            # make message search case insensitive
+            return Condition(
+                Function("positionCaseInsensitive", [self.builder.column("message"), value]),
+                Op.NEQ if search_filter.operator in EQUALITY_OPERATORS else Op.EQ,
+                0,
+            )
+
+    def _project_slug_filter_converter(self, search_filter: SearchFilter) -> Optional[WhereType]:
+        return filter_aliases.project_slug_converter(self.builder, search_filter)
+
+    @property
+    def field_alias_converter(self) -> Mapping[str, Callable[[str], SelectType]]:
+        return {
+            PROJECT_ALIAS: self._resolve_project_slug_alias,
+            PROJECT_NAME_ALIAS: self._resolve_project_slug_alias,
+        }
+
+    def _resolve_project_slug_alias(self, alias: str) -> SelectType:
+        return field_aliases.resolve_project_slug_alias(self.builder, alias)
+
+    @property
+    def function_converter(self) -> Mapping[str, SnQLFunction]:
+        return {
+            function.name: function
+            for function in [
+                SnQLFunction(
+                    "count",
+                    snql_aggregate=lambda _, alias: Function(
+                        "countMerge",
+                        [SnColumn("count")],
+                        alias,
+                    ),
```

Comment:
Not sure if its different for the Function dataset, but for the Metrics dataset snuba will add the `-Merge` and the column for you, so doing this might work too
```suggestion
                        "count",
                        [],
                        alias,
                    ),
```

Reply:
I think we can drop the `-Merge` part
```suggestion
                        "count",
                        [SnColumn("count")],
                        alias,
                    ),
```
I don't recall implementing anything in snuba to autofill in the argument for us